PaddleCV/Research/webvision2018/infer.py

```diff
@@ -0,0 +1,119 @@
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import time
+import sys
+import math
+import numpy as np
+import argparse
+import functools
+
+import paddle
+import paddle.fluid as fluid
+import reader_cv2 as reader
+import models
+import utils
+from utils.utility import add_arguments,print_arguments
+
+
+parser = argparse.ArgumentParser(description=__doc__)
+# yapf: disable
+add_arg = functools.partial(add_arguments, argparser=parser)
+add_arg('use_gpu',          bool, True,                 "Whether to use GPU or not.")
+add_arg('class_dim',        int,  5000,                 "Class number.")
+add_arg('image_shape',      str,  "3,224,224",          "Input image size")
+add_arg('with_mem_opt',     bool, True,                 "Whether to use memory optimization or not.")
+add_arg('pretrained_model', str,  None,                 "Whether to use pretrained model.")
+add_arg('model',            str,  "ResNeXt101_32x4d",   "Set the network to use.")
+add_arg('save_inference',   bool, False,                "Whether to save inference model or not")
+add_arg('resize_short_size', int, 256,                  "Set resize short size")
+add_arg('img_list',          str, None,                 "list of valset")
+add_arg('img_path',          str, None,                 "path of valset")
+# yapf: enable
+
+def infer(args):
+    # parameters from arguments
+    class_dim = args.class_dim
+    model_name = args.model
+    save_inference = args.save_inference
+    pretrained_model = args.pretrained_model
+    with_memory_optimization = args.with_mem_opt
+    image_shape = [int(m) for m in args.image_shape.split(",")]
+    model_list = [m for m in dir(models) if "__" not in m]
+    assert model_name in model_list, "{} is not in lists: {}".format(args.model,
+                                                                     model_list)
+
+    image = fluid.layers.data(name='image', shape=image_shape, dtype='float32')
+
+    # model definition
+    model = models.__dict__[model_name]()
+    if model_name == "GoogleNet":
+        out, _, _ = model.net(input=image, class_dim=class_dim)
+    else:
+        out = model.net(input=image, class_dim=class_dim)
+        #out = fluid.layers.softmax(out)
+
+    test_program = fluid.default_main_program().clone(for_test=True)
+
+    fetch_list = [out.name]
+    if with_memory_optimization and not save_inference:
+        fluid.memory_optimize(
```

Comment:
memory_optimize is deprecated and there are several methods in https://www.paddlepaddle.org.cn/documentation/docs/zh/1.5/advanced_usage/best_practice/memory_optimize.html

Reply:
I have remove these lines.